packages/datadog-instrumentations/src/http2/server.js

```diff
@@ -51,18 +47,19 @@ function wrapEmit (emit) {
     if (eventName === 'request') {
       res.req = req
 
-      const asyncResource = new AsyncResource('bound-anonymous-fn')
-      return asyncResource.runInAsyncScope(() => {
-        startServerCh.publish({ req, res })
-
-        shimmer.wrap(res, 'emit', wrapResponseEmit)
+      const ctx = { req, res }
+      return startServerCh.runStores(ctx, () => {
+        shimmer.wrap(res, 'emit', emit => responseCh.runStores(ctx, () => {
```

Comment:
This should happen at emit time and not at wrap time. From my understanding of the new code it should look something like this:

```js
function wrapResponseEmit (emit, ctx) {
  return function (eventName, event) {
    ctx.req = this.req
    ctx.eventName = eventName

    return eventName === 'close' && finishServerCh.hasSubscribers
      ? finishServerCh.runStores(ctx, emit, this, ...arguments)
      : responseCh.runStores(ctx, emit, this, ...arguments)
  }
}
```

Reply:
Fixed in our review session.